src/lib/y2users/config_element/base.rb

```diff
@@ -0,0 +1,106 @@
+# Copyright (c) [2021] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+module Y2Users
+  module ConfigElement
+    # Base class for an element that can be attached to a config
+    class Base
```

Comment:
I have a doubt about the namespaces. Now we have:

~~~
Y2Users::Config
Y2Users::User < Y2Users::ConfigElement::Base
Y2Users::Group < Y2Users::ConfigElement::Base
Y2Users::UsersCollection < Y2Users::ConfigElement::Collection
Y2Users::GroupsCollection < Y2Users::ConfigElement::Collection
~~~

But maybe `User` and `Group` should be below `ConfigElement`:

~~~
Y2Users::Config
Y2Users::ConfigElement::User < Y2Users::ConfigElement::Base
Y2Users::ConfigElement::Group < Y2Users::ConfigElement::Base
Y2Users::ConfigElement::Collection::Users < Y2Users::ConfigElement::Collection::Base
Y2Users::ConfigElement::Collection::Groups < Y2Users::ConfigElement::Collection::Base
~~~

Or maybe everything could be at the same level:

~~~
Y2Users::Config
Y2Users::User < Y2Users::ConfigElement
Y2Users::Group < Y2Users::ConfigElement
Y2Users::UsersCollection < Y2Users::ConfigElementCollection
Y2Users::GroupsCollection < Y2Users::ConfigElementCollection
~~~

Reply:
At first sight, from the POV of a user of our API, I would say that `ConfigElement` is an implementation detail, so I do not think it should be in a prominent place like option 2.

If we plan to have more "config elements", I would go with option 3. Otherwise, I would keep option 1.

Reply:
I prefer option 3 too

Reply:
Thanks guys! Let's go with option 3.